tests/unit/limitations_test.c

```diff
@@ -280,32 +285,21 @@ CTEST2(limitations, test_splinterdb_created_wo_background_threads)
    default_data_config_init(TEST_MAX_KEY_SIZE, &default_data_cfg);
    create_default_cfg(&cfg, &default_data_cfg);
 
-   // No background threads should be created with this configuration.
-   cfg.num_normal_bg_threads   = 2;
-   cfg.num_memtable_bg_threads = 0;
+   // Task system should be setup with background threads
+   cfg.num_normal_bg_threads   = 1;
+   cfg.num_memtable_bg_threads = 1;
 
    int rc = splinterdb_create(&cfg, &kvsb);
    ASSERT_EQUAL(0, rc);
 
-   ASSERT_FALSE(splinterdb_task_system_uses_bg_threads(kvsb));
-   splinterdb_close(&kvsb);
-
-   // Do the check for the reverse configuration.
-   cfg.num_normal_bg_threads   = 0;
-   cfg.num_memtable_bg_threads = 2;
-
-   rc = splinterdb_create(&cfg, &kvsb);
-   ASSERT_EQUAL(0, rc);
-
-   ASSERT_FALSE(splinterdb_task_system_uses_bg_threads(kvsb));
    splinterdb_close(&kvsb);
 }
 
 /*
- * Test that SplinterDB tasks system requires non-zero config for both types
- * of task groups for background threads to be created.
+ * Test that SplinterDB can be created even when background threads use
+ * up all the slots.
  */
-CTEST2(limitations, test_splinterdb_created_w_background_threads)
+CTEST2(limitations, test_splinterdb_created_w_all_background_threads)
```

Comment:
I am reworking this newly added case slightly to create 62 bg-threads. Then, test case will show that we can create one more user thread but the next user-thread creation should fail. (Am debugging that logic right now... )